src/shared/types/index.ts

```diff
@@ -29,6 +30,95 @@ interface StreamLedger {
   type: string
 }
 
+interface LedgerEntryWarning {
+  id: number
+  message: string
+}
+
+interface LedgerEntryMajority {
+  Majority: {
+    Amendment: string
+    CloseTime: number
+  }
+}
+
+interface LedgerEntryAmendmentsResponse {
+  result: {
+    index: string
+    ledger_hash: string
+    ledger_index: number
+    node: {
+      Amendments: string[]
+      Flags: number
+      LedgerEntryType: string
+      Majorities: LedgerEntryMajority[]
+      index: string
+    }
+    validated: boolean
+    warnings: LedgerEntryWarning[]
+  }
+  status: string
+  type: string
+}
+
+interface LedgerEnableAmendmentResponse {
+  result: {
+    ledger: {
+      accepted: boolean
+      account_hash: string
+      close_flags: number
+      close_time: number
+      close_time_human: number
+      close_time_resolution: number
+      closed: boolean
+      hash: string
+      ledger_hash: string
+      ledger_index: string
+      parent_close_time: number
+      parent_hash: string
+      seqNum: string
+      totalCoins: string
+      total_coins: string
+      transaction_hash: string
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any -- Use any since we do not need to build Transaction model.
+      transactions: any[]
```

Comment:
Why not use types in xrpl.js?

Reply:
Currently, the `Transaction` type in xrpl.js is missing `EnableAmendment` tx type:
https://github.com/XRPLF/xrpl.js/blob/main/packages/xrpl/src/models/transactions/transaction.ts#L90-L129
I will create a PR to update it and add a TODO to update the code here accordingly once the change has been released